### DISCUSSION TITLE
How to: Automatic html page loading/reloading (local only)

### DISCUSSION THREAD

**Discussion by heysokam** [zigzap/zap#114] [Q&A]
Hi there
I'm searching for  a way to create a simple localdev server that automatically reloads html pages when they change on disk, using zig.
The goal is to create something similar to https://marketplace.visualstudio.com/items?itemName=yandeu.five-server

> Note: To narrow it further, the goal is not to make a production-ready server.
> Just a localdev-only server that serves all files from a folder and automatically refreshes when any file changes, that's all.
> Similar to how `npm run serve` or `docusaurus start` would work, thanks to their webpack local backend.

I have no experience creating an http server, so I have some questions:
- Is there a way to create this minimalistic localdev html serving behavior using `zap`?
- If yes, which of the examples would be the closest to this idea?
- If there is no examples, does zap have the tools to create such a localdev server, or am I looking at the wrong tool for the right job? :thinking:

**Comment by renerocksai** [comment]
The most basic example will do. Use the public folder feature. 

If you want to serve files yourself, make sure to set all caching related headers so that caching in the browser is prevented. 

Sorry, I am at an airport and can't do your work for you. But I hope my pointers help anyway.

**Reply by heysokam** [reply]
@renerocksai No worries. I will give it a try.
I just wanted to make sure that zap can achieve the task before giving it an attempt :writing_hand:

I will try to make it work, and will send further questions here if I encounter any issues :ok_hand:
